src/pattern-library/components/Library.tsx

```diff
@@ -68,78 +45,115 @@ function Page({ children, intro, title }: LibraryPageProps) {
   );
 }
 
-export type LibrarySectionProps = {
+export type LibraryHeadingProps = JSX.HTMLAttributes<HTMLHeadingElement> & {
+  children: ComponentChildren;
+  level: number;
+};
+
+function Heading({
+  children,
+  level = 2,
+  ...htmlAttributes
+}: LibraryHeadingProps) {
+  if (level <= 2) {
+    return (
+      <h2 className="text-3xl text-slate-600 font-bold" {...htmlAttributes}>
+        {children}
+      </h2>
+    );
+  } else if (level === 3) {
+    return (
+      <h3 className="text-2xl text-slate-600 font-medium" {...htmlAttributes}>
+        {children}
+      </h3>
+    );
+  } else {
+    return (
+      <h4
+        className="text-lg border-b border-stone-300 text-slate-600 font-normal"
+        {...htmlAttributes}
+      >
+        {children}
+      </h4>
+    );
+  }
+}
+
+export type LibrarySectionProps = LibraryElementAttributes & {
   children?: ComponentChildren;
   intro?: ComponentChildren;
-  id?: string;
   title?: string;
+  level?: number;
 };
 
+// Keep track of <Section> nested depth
+const SectionDepthContext = createContext<number>(1);
+
 /**
  * Render a primary section of a page. Each component documented on a pattern-
  * library page gets its own section.
  */
-function Section({ children, id, intro, title }: LibrarySectionProps) {
+function Section({
+  children,
+  intro,
+  level,
+  title,
+  ...htmlAttributes
+}: LibrarySectionProps) {
+  const sectionDepth = useContext(SectionDepthContext);
+  const depth = level ?? sectionDepth + 1;
```

Comment:
So: If provided with a `level` prop it will render at that level, otherwise it will automatically set depth based on parent depth.

Note to self, mostly: If you provide a `level`, that will effectively reset the depth to that `level` for descendants, but that's probably desirable anyway...